xarray/tests/test_backends.py

```diff
@@ -992,6 +1032,80 @@ def test_encoding_unlimited_dims(self):
             with pytest.warns(UserWarning):
                 ds.to_netcdf(tmp_file, engine='h5netcdf', unlimited_dims=['y'])
 
+# tests pending h5netcdf fix
+#class H5NetCDFDataTestAutocloseTrue(H5NetCDFDataTest):
+#    autoclose = True
+
+class OpenMFDatasetTest(TestCase):
+    autoclose = True
+    def validate_open_mfdataset_autoclose(self, engine, nfiles=10):
+        randdata = np.random.randn(nfiles)
+        original = Dataset({'foo': ('x', randdata)})
+        # test standard open_mfdataset approach with too many files
+        with create_tmp_files(nfiles) as tmpfiles:
+            for readengine in engine:
+                writeengine = readengine if readengine != 'pynio' else 'netcdf4'
+                # split into multiple sets of temp files
+                for ii in original.x.values:
+                    (
+                     original.isel(x=slice(ii, ii+1))
```

Comment:
I think this would be slightly more readable with an intermediate variable, or removing the line break on the line above.